File: LC/java/Remove Duplicates from Sorted List .java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode curr = head;
        
        if(head == null || head.next == null){
            return head;
        }
        while(curr.next != null){
            
            if(curr.next.val != curr.val){
                curr=curr.next;
            }
            else{
                curr.next = curr.next.next;
            }
        }
        return head;
    }
}

File: LC/python/Remove Duplicates from Sorted List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param head, a ListNode
    # @return a ListNode
    def deleteDuplicates(self, head):
        if head == None or head.next == None:
            return head
        
        temp = head
        
        while temp.next:
            if temp.val == temp.next.val:
                temp.next = temp.next.next
            else:
                temp = temp.next
        
        return head
